inst/unitTests/runit.xts.R

```diff
@@ -50,3 +50,52 @@ test..xts_dimnames_in_dots <- function() {
   y <- xts(1:5, index(x), dimnames = list(NULL, "x"))
   checkEquals(x, y)
 }
+
+checkXtsClass <- function(xts, class) {
+  checkEquals(attr(xts, "tclass"), class)
+  checkEquals(attr(xts, ".indexCLASS"), class)
+  checkEquals(attr(attr(xts, "index"), "tclass"), class)
+}
+
+### Check that .indexCLASS takes precedence over tclass when both specified
+test..xts_class <- function() {
+  checkXtsClass(.xts(1, 1), c("POSIXct", "POSIXt"))
+  checkXtsClass(.xts(1, 1, tclass="timeDate"), "timeDate")
+  checkXtsClass(.xts(1, 1, .indexCLASS="Date"), "Date")
+  checkXtsClass(.xts(1, 1, tclass="timeDate", .indexCLASS="Date"), "Date")
+
+  ## also check that tclass is ignored if specified as part of index
+  checkXtsClass(.xts(1, structure(1, tzone="",tclass="yearmon")), c("POSIXct", "POSIXt"))
+  checkXtsClass(.xts(1, structure(1, tzone="",tclass="yearmon"), tclass="timeDate"), "timeDate")
+  checkXtsClass(.xts(1, structure(1, tzone="",tclass="yearmon"), .indexCLASS="Date"), "Date")
+  checkXtsClass(.xts(1, structure(1, tzone="",tclass="yearmon"), tclass="timeDate", .indexCLASS="Date"), "Date")
+}
+
+checkXtsTz <- function(xts, tzone, .indexTZ) {
```

Comment:
This test looks for attributes on specific objects, but it probably should use the extractor functions `indexTZ()` and `tzone()`.